plugin/clone/src/clone_plugin.cc

```diff
@@ -517,6 +520,36 @@ static int plugin_clone_remote_client(THD *thd, const char *remote_host,
 @param[in]	socket	network socket to remote client
 @return error code */
 static int plugin_clone_remote_server(THD *thd, MYSQL_SOCKET socket) {
+  int dscp_val = clone_dscp_on_socket;
+
+  bool set_dscp = false;
+  if (dscp_val != 0) {
+    set_dscp = true;
+  }
+
+  if (set_dscp) {
+    int tos = dscp_val << 2;
+    uint16_t test_family;
+    socklen_t len = sizeof(test_family);
+    int res = mysql_socket_getsockopt(socket, SOL_SOCKET, SO_DOMAIN, reinterpret_cast<void *>(&test_family), &len);
+
+    if (res != 0) {
+    // NO_LINT_DEBUG
+      my_message(ER_CLONE_DONOR, "Failed to get socket domain while adjusting DSCP_QOS on donor", MYF(0));
```

Comment:
I believe all these `my_message` calls should be something else, e.g.  `my_error` followed by `LogPluginErr(ERROR_LEVEL, ER_CLONE_CLIENT_TRACE, ... )` - I'd suggest to simulate a failure here and see what happens, both what is returned to the client and what is printed to the error log.